apps/staking/src/api.ts

```diff
@@ -144,73 +144,114 @@ export const getStakeAccounts = async (
-        cooldown2: filterOISPositions(publisher, PositionState.UNLOCKED),
+  if (stakeAccount === undefined) {
+    const [integrityStakingPublishers, ownerAtaAccount] = await Promise.all([
+      loadPublisherData(client),
+      client.getOwnerPythAtaAccount(),
+    ]);
+
+    return {
+      lastSlash: undefined,
+      availableRewards: 0n,
+      expiringRewards: undefined,
+      total: 0n,
+      governance: {
+        warmup: 0n,
+        staked: 0n,
+        cooldown: 0n,
+        cooldown2: 0n,
       },
-    })),
-  };
+      unlockSchedule: [],
+      locked: 0n,
+      walletAmount: ownerAtaAccount.amount,
+      integrityStakingPublishers,
+    };
+  } else {
+    const [
+      stakeAccountCustody,
+      publishers,
+      ownerAtaAccount,
+      currentEpoch,
+      unlockSchedule,
+    ] = await Promise.all([
+      client.getStakeAccountCustody(stakeAccount.address),
+      loadPublisherData(client),
+      client.getOwnerPythAtaAccount(),
+      getCurrentEpoch(client.connection),
+      client.getUnlockSchedule(stakeAccount.address),
+    ]);
+
+    const filterGovernancePositions = (positionState: PositionState) =>
+      getAmountByTargetAndState({
+        stakeAccountPositions: stakeAccount,
+        targetWithParameters: { voting: {} },
+        positionState,
+        epoch: currentEpoch,
+      });
+
+    const filterOISPositions = (
+      publisher: PublicKey,
+      positionState: PositionState,
+    ) =>
+      getAmountByTargetAndState({
+        stakeAccountPositions: stakeAccount,
+        targetWithParameters: { integrityPool: { publisher } },
+        positionState,
+        epoch: currentEpoch,
+      });
+
+    return {
+      lastSlash: undefined, // TODO
+      availableRewards: 0n, // TODO
+      expiringRewards: undefined, // TODO
+      total: stakeAccountCustody.amount,
+      governance: {
+        warmup: filterGovernancePositions(PositionState.LOCKING),
+        staked: filterGovernancePositions(PositionState.LOCKED),
+        cooldown: filterGovernancePositions(PositionState.PREUNLOCKING),
+        cooldown2: filterGovernancePositions(PositionState.UNLOCKED),
+      },
+      unlockSchedule,
+      locked: unlockSchedule.reduce((sum, { amount }) => sum + amount, 0n),
+      walletAmount: ownerAtaAccount.amount,
+      integrityStakingPublishers: publishers.map((publisher) => ({
+        ...publisher,
+        positions: {
+          warmup: filterOISPositions(
+            publisher.publicKey,
+            PositionState.LOCKING,
+          ),
+          staked: filterOISPositions(publisher.publicKey, PositionState.LOCKED),
+          cooldown: filterOISPositions(
+            publisher.publicKey,
+            PositionState.PREUNLOCKING,
+          ),
+          cooldown2: filterOISPositions(
+            publisher.publicKey,
+            PositionState.UNLOCKED,
+          ),
+        },
+      })),
+    };
+  }
+};
+
+const loadPublisherData = async (client: PythStakingClient) => {
```

Comment:
@keyvankhademi FYI, this stuff is separated out because there are now two separate flows for fetching dashboard data -- one for when there is a stake account and one for when there isn't.  The publisher data is common between the two.